server/src/repositories/user.repo.ts

```diff
@@ -0,0 +1,96 @@
+import ApiError from "../error/api.error.js";
+import { UserEntity } from "../db/entities/user.entity.js";
+import dbConnector from "../db/db.connector.js";
+
+export const UserRepo = dbConnector.getRepository(UserEntity).extend({
+  async createUser(
```

Comment:
it's better to wrap parameters into an object and pass only one variable
data: {
name: string,
    surname: string,
    DoB: string,
    email: string,
    password: string,
    roleId: string
}